#goal definition
string function_name
KeyValue[] function_params
---
#result definition
int32 error_code
string message
---
#feedback definition
string message
